runtime/src/bank/fee_distribution.rs

```diff
@@ -161,53 +212,40 @@ impl Bank {
                     rent_share
                 };
                 if !enforce_fix || rent_to_be_paid > 0 {
-                    let mut account = self
-                        .get_account_with_fixed_root(&pubkey)
-                        .unwrap_or_default();
-                    let rent = self.rent_collector().rent;
-                    let recipient_pre_rent_state = RentState::from_account(&account, &rent);
-                    let distribution = account.checked_add_lamports(rent_to_be_paid);
-                    let recipient_post_rent_state = RentState::from_account(&account, &rent);
-                    let rent_state_transition_allowed = recipient_post_rent_state
-                        .transition_allowed_from(&recipient_pre_rent_state);
-                    if !rent_state_transition_allowed {
-                        warn!(
-                            "Rent distribution of {rent_to_be_paid} to {pubkey} results in \
-                            invalid RentState: {recipient_post_rent_state:?}"
-                        );
-                        datapoint_warn!(
-                            "bank-rent_distribution_invalid_state",
-                            ("slot", self.slot(), i64),
-                            ("pubkey", pubkey.to_string(), String),
-                            ("rent_to_be_paid", rent_to_be_paid, i64)
-                        );
```

Comment:
Just calling out that we do lose this warning and datapoint in the new `deposit_fees`. Hopefully, `prevent_rent_paying_rent_recipients()` will be active before this PR is released, so we'll no longer care about tracking that specific case so carefully.

Reply:
Yeah I was curious if you wanted to keep that or not, I can keep it in if you want!

Reply:
@jeffwashington , @brooksprumo , I think you're the main users of this datapoint currently. Thoughts on removing it?

Reply:
The latest patch of this PR removes this particular datapoint_warn but has two datapoint_warn's that track how many lamports were burned during both tx and rent fee distribution

```rust
datapoint_warn!(
    "bank-burned_rent",
    ("slot", self.slot(), i64),
    ("num_lamports", rent_to_burn, i64)
);

datapoint_warn!(
    "bank-burned_fee",
    ("slot", self.slot(), i64),
    ("num_lamports", deposit, i64),
    ("error", err.to_string(), String),
);
```

Reply:
For `bank-burned_fee` we track the slot which allows us to infer who the leader was that failed to collect tx fees